azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -546,6 +556,30 @@ def __try_set_cpu_quota(quota):  # pylint: disable=unused-private-member
                 return False
             return True
 
+        def _check_processes_in_agent_cgroup_before_enable(self, agent_slice):
+            """
+            This check ensures that before we enable the agent's cgroups, there are no unexpected processes in the agent's cgroup already.
+
+            The issue we observed that long running extension processes may be in agent cgroups if agent goes this cycle enabled(1)->disabled(2)->enabled(3).
+            1. Agent cgroups enabled in some version
+            2. Disabled agent cgroups due to check_cgroups regular check, now extension runs will be in agent cgroups.
+            3. When ext_hanlder restart and enable the cgroups again, already running processes from step 2 still be in agent cgroups. This may cause the extensions run with agent limit.
+            """
+            if agent_slice not in AZURE_SLICE:
+                return False
+
+            if self._agent_cpu_cgroup_path is None:
```

Comment:
Why is this check here?

Reply:
we use this path for finding processes in cgroup.procs file. It could be none for v2 or

Reply:
Why do we use agent_cpu_cgroup_path to check which processes in agent cgroup? Why not memory_cgroup_path? or should we be checking both cpu and memory paths?

it's possible for memory to be mounted/in use but cpu not